general/communication.hpp

```diff
@@ -76,13 +79,23 @@ class Mpi
    static void Init_(int *argc, char ***argv)
    {
       MFEM_VERIFY(!IsInitialized(), "MPI already initialized!")
+#ifndef MFEM_USE_JIT
       MPI_Init(argc, argv);
+#else
+      Jit::Init(argc, argv);
+#endif
```

Comment:
We can't rely that apps will use our `mfem::Mpi::Init`, so we may want to rework JIT initialization to make it independent of `mfem::Mpi::Init`.